topsort-api-v1.yml

```diff
@@ -214,7 +214,10 @@ components:
         - location
       properties:
         location:
-          description: The location this banner is being run for.
+          description: >
+            By using this option, you will filter products that belong to campaigns that have the same location configuration.
```

Comment:
I'm not sure we want to tell them they can use this for anything that's not strictly a geographical location (special placements on the App, such as checkout, cross-sell, swimlanes, etc).

In banners the slot configuration acts as the page location option.
Also, if they do that there's gonna be no way of running an auction for "users in turkey after checkout".
The location field has the problem that increments the number of bids. Let's say they have 21 locations and the vendors don't want to appear in one, then we would require to create 20 bids for each product being promoted.


Also the documentation should say that for this to work they need to configure the available locations in the manager.

Reply:
@sk- AFAIK customers hack this field to be able to run multiple targeting like `turkey:checkout`

Reply:
We shouldn't be allowing that because that would mean a quadratic number of locations, and also the manager would provide a suboptimal experience as it was designed assuming this field is an actual location

Cc @wicha

Reply:
Even if they are using the API, they should create the locations. And we should validate the locations at campaign creation time.

The fact that they are abusing the API, does not mean that we need to document those hacks in the documentation.

Glovo is currently operating in over 1300 cities and 25 markets, giving an average of 52 cities per market. If some of their bids are built for all those locations, we will have an increase in bids for which the system was not designed.

Location was a quick hack for airlift but always thinking that there was going to have a relatively low number of locations.

Reply:
location as Geolocation is an important use case for Glovo, if our current implementation is not the best we can definitely plan to improve it. 
About the creative ways of using location, as App features and placements, its also an use case, that even our competitors have, like CitrusAd.

We can hold the documentation update until we are sure we can handle it, but we should be ready to support this use case.